File: 01813.cpp
#include <iostream>
using namespace std;

int trues[100001];

int main() {
	int N; cin >> N;

	for (int i = 0; i < N; i++) {
		int a; cin >> a;
		trues[a]++;
	}

	int cnt = 0;
	for (int i = 100000; i >= 0; i--) {
		if (trues[i] == i) {
			cout << i << endl;
			return 0;
		}
	}

	cout << -1 << endl;
}

File: 01813.py
trues = [0] * 100001
N = int(input())

a = list(map(int, input().split()))
for n in a:
    trues[n] += 1

cnt = 0
for i in range(100000, -1, -1):
    if trues[i] == i:
        print(i)
        break
else: print(-1)
